From 520908b47fb25ff6b7b13d71cc4d6626f8d0c229 Mon Sep 17 00:00:00 2001
From: Adrien Ricciardi <aricciardi@baylibre.com>
Date: Mon, 18 Sep 2023 20:18:28 +0200
Subject: [PATCH 04/38] os: Fixed an unused parameter warning in
 CALL_SHUTDOWN_HOOK().

The 'error' parameter is not used when the shutdown hook support is not enabled
and triggers a warning in and triggers a compilation warning in os/tpl_os_os_kernel.c,
function tpl_call_shutdown_os().

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>
---
 os/tpl_os_hooks.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os/tpl_os_hooks.h b/os/tpl_os_hooks.h
index cbe3c122..f4e4b5ee 100644
--- a/os/tpl_os_hooks.h
+++ b/os/tpl_os_hooks.h
@@ -112,7 +112,7 @@
     ShutdownHook(error); \
     RESET_RUNNING_TRUSTED
 #else
-#   define CALL_SHUTDOWN_HOOK(error)
+#   define CALL_SHUTDOWN_HOOK(error) { (void) error; }
 #endif
 
 /**
-- 
2.34.1

